File: 162.py
from typing import List

class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        n = len(nums)
        def get(i: int) -> int:
            if i == -1 or i == n:
                return float('-inf')
            return nums[i]
        left, right, res = 0, n - 1, -1
        while left <= right:
            mid = (left + right) // 2
            if get(mid - 1) < get(mid) > get(mid + 1):
                res = mid
                break
            elif get(mid) < get(mid + 1):
                left = mid + 1
            elif get(mid) < get(mid - 1):
                right = mid - 1
        return res

File: go/162.go
package main

import "math"

func findPeakElement(nums []int) int {
	n := len(nums)
	var get func(i int) int
	get = func(i int) int {
		if i == -1 || i == n {
			return math.MinInt64
		}
		return nums[i]
	}
	l, r := 0, n-1
	for l <= r {
		mid := (l + r) / 2
		if get(mid-1) < nums[mid] && get(mid+1) < nums[mid] {
			return mid
		}
		if get(mid-1) >= nums[mid] {
			r = mid - 1
		} else {
			l = mid + 1
		}
	}
	return -1
}
